tests/smart/test_smartdevice.py

```diff
@@ -763,3 +773,218 @@ class DummyModule(SmartModule):
     )
     mod = DummyModule(dummy_device, "dummy")
     assert mod.query() == {}
+
+
+@hub_all
+@pytest.mark.xdist_group(name="caplog")
+@pytest.mark.requires_dummy
+async def test_dynamic_devices(dev: Device, caplog: pytest.LogCaptureFixture):
+    """Test dynamic child devices."""
+    if not dev.children:
```

Comment:
It'd be great if this would be split up into multiple tests, to avoid mixing additions and deletions.

Reply:
Fair enough :+1: